test/utils/readRowsImpl.ts

```diff
@@ -228,7 +231,60 @@ export function readRowsImpl(
     });
 
     let chunksSent = 0;
```

Comment:
The entire block of changes shown below is copy/pasted from ReadRowsImpl2.ts. The next PR is going to get rid of the repetition in these if statements.